internal/querynode/flow_graph_delete_node.go

```diff
@@ -96,16 +97,20 @@ func (dNode *deleteNode) Operate(in []flowgraph.Msg) []flowgraph.Msg {
 		)
 
 		if dNode.replica.getSegmentNum() != 0 {
-			processDeleteMessages(dNode.replica, delMsg, delData)
+			err := processDeleteMessages(dNode.replica, delMsg, delData)
+			if err != nil {
+				// error occurs when missing meta info or unexpected pk type, should not happen
+				panic(fmt.Errorf("deleteNode processDeleteMessages failed, collectionID = %d, err = %s", delMsg.CollectionID, err))
```

Comment:
How ever, I don't like the style of processDeleteMessages, where we handle partition not exist, segment not exist......
All of these check should happen once and only once

Reply:
I'll handle it in another pr